Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/CertficateBean.java

```diff
@@ -534,4 +559,143 @@ private static boolean checkCompletedChain(Map<String, CertificateDTO> chain) {
+        return referenceNumber;
+    }
+
+    public void setReferenceNumber(String referenceNumber) {
+        this.referenceNumber = referenceNumber;
+    }
+
+    public String getOrganizationalUnit() {
+        return organizationalUnit;
+    }
+
+    public void setOrganizationalUnit(String organizationalUnit) {
+        this.organizationalUnit = organizationalUnit;
+    }
+
+    public String getOrganization() {
+        return organization;
+    }
+
+    public void setOrganization(String organization) {
+        this.organization = organization;
+    }
+
+    public String getCountryName() {
+        return countryName;
+    }
+
+    public void setCountryName(String countryName) {
+        this.countryName = countryName;
+    }
+
+    public int getImportWizardTabIndex() {
+        return importWizardTabIndex;
+    }
+
+    public void setImportWizardTabIndex(int importWizardTabIndex) {
+        this.importWizardTabIndex = importWizardTabIndex;
+    }
+
+    public void createCertificate() {
+        if (StringUtils.isBlank(getAlias()) || StringUtils.isBlank(getReferenceNumber())
+            || StringUtils.isBlank(getOrganizationalUnit()) || StringUtils.isBlank(getOrganization())
+            || StringUtils.isBlank(getCountryName())) {
+            HelperUtil.addMessageError(null, "Required fields cannot be empty when creating certificate.");
+        }
+
+        boolean status = service.createCertificate(getAlias(), getReferenceNumber(), getOrganizationalUnit(),
+            getOrganization(), getCountryName());
+        if (status) {
+            HelperUtil.addMessageInfo(null,
+                MessageFormat.format("Successfully created certifcate for: {0}", getAlias()));
+            importWizardTabIndex = TABINDEX_CREATECSR;
+        } else {
+            HelperUtil.addMessageError(null, MessageFormat.format("Failed to create certifcate for: {0}", getAlias()));
+        }
+    }
+
+    public void createCSR() {
+        if (StringUtils.isBlank(getAlias())) {
+            HelperUtil.addMessageError(null, "Required fields cannot be empty when creating CSR.");
+            return;
+        }
+        csrFile = null;
+
+        SimpleCertificateResponseMessageType response = service.createCSR(getAlias());
+
+        if (null != response && response.isStatus()) {
+            HelperUtil.addMessageInfo(null,
+                MessageFormat.format("Successfully created CSR for: {0}", getAlias()));
+            if(null != response.getCsrData()){
+                DataHandler data = response.getCsrData();
+                String filename = MessageFormat.format("{0}-{1}.csr", getAlias(), CoreHelpUtils.formatDate("yyyyMMdd", new Date()));
+                try {
+                    csrFile = new DefaultStreamedContent(data.getInputStream(), data.getContentType(), filename);
+                } catch (IOException e) {
+                    LOG.error("unable to open input-stream from response", e.getLocalizedMessage(), e);
```

Comment:
shouldn't we let the user know that the operation failed here?

Reply:
for the moment no the user doesn't not need to know the input-stream fail to create; this is for the step-3